pdsClient/pdsClient.go

```diff
@@ -1,110 +1,110 @@
-    err := json.NewEncoder(&buf).Encode(&params)
-    if err != nil {
-        return result, err
-    }
-    request, err := http.NewRequest("PUT", c.Host+"/"+PDSServiceBasePath+"/access_keys"+fmt.Sprintf("/%s", params.WriterID)+fmt.Sprintf("/%s", params.UserID)+fmt.Sprintf("/%s", params.ReaderID)+fmt.Sprintf("/%s", params.RecordType), &buf)
-    if err != nil {
-        return result, err
-    }
-    err = e3dbClients.MakeE3DBServiceCall(c.E3DBAuthClient, ctx, request, &result)
-    return result, err
+	var result *PutAccessKeyResponse
+	var buf bytes.Buffer
+	err := json.NewEncoder(&buf).Encode(&params)
+	if err != nil {
+		return result, err
+	}
+	request, err := http.NewRequest("PUT", c.Host+"/"+PDSServiceBasePath+"/access_keys"+fmt.Sprintf("/%s", params.WriterID)+fmt.Sprintf("/%s", params.UserID)+fmt.Sprintf("/%s", params.ReaderID)+fmt.Sprintf("/%s", params.RecordType), &buf)
+	if err != nil {
+		return result, err
+	}
+	err = e3dbClients.MakeE3DBServiceCall(c.E3DBAuthClient, ctx, request, &result)
+	return result, err
 }
 
 // WriteRecord attempts to store a record in e3db, returning stored record and error (if any).
 // XXX: Data is not encrypted in this method, it is the caller's responsibility to ensure data is encrypted before sending to e3db.
 func (c *E3dbPDSClient) WriteRecord(ctx context.Context, params WriteRecordRequest) (*WriteRecordResponse, error) {
-    var result *WriteRecordResponse
-    var buf bytes.Buffer
-    err := json.NewEncoder(&buf).Encode(&params)
-    if err != nil {
-        return result, err
-    }
-    request, err := http.NewRequest("POST", c.Host+"/"+PDSServiceBasePath+"/records", &buf)
-    if err != nil {
-        return result, err
-    }
-    err = e3dbClients.MakeE3DBServiceCall(c.E3DBAuthClient, ctx, request, &result)
-    return result, err
+	var result *WriteRecordResponse
+	var buf bytes.Buffer
+	err := json.NewEncoder(&buf).Encode(&params)
+	if err != nil {
+		return result, err
+	}
+	request, err := http.NewRequest("POST", c.Host+"/"+PDSServiceBasePath+"/records", &buf)
+	if err != nil {
+		return result, err
+	}
+	err = e3dbClients.MakeE3DBServiceCall(c.E3DBAuthClient, ctx, request, &result)
+	return result, err
 }
 
 // ListRecords returns a list of records using any filters provided as params, and error (if any).
-func (c *E3dbPDSClient) ListRecords(ctx context.Context, params ListRecordsRequest) (*ListRecordsResult, error) {
-    var result *ListRecordsResult
-    var buf bytes.Buffer
-    err := json.NewEncoder(&buf).Encode(&params)
-    if err != nil {
-        return result, err
-    }
-    request, err := http.NewRequest("POST", c.Host+"/"+PDSServiceBasePath+"/search", &buf)
-    if err != nil {
-        return result, err
-    }
-    err = e3dbClients.MakeE3DBServiceCall(c.E3DBAuthClient, ctx, request, &result)
-    return result, err
+func (c *E3dbPDSClient) ListRecords(ctx context.Context, authToken string, params ListRecordsRequest) (*ListRecordsResult, error) {
```

Comment:
This should be a separate method, we want to have one method that "normal" clients can use and one for proxy clients(like the executor) to use.

Reply:
Do the tests still pass with this update as is?

Reply:
It does not (my mistake I did not commit updated test), but I have separated the methods and added line to test both methods.

Reply:
Awesome!